components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -474,6 +513,9 @@ class SystemEngineView @JvmOverloads constructor(
     override fun canScrollVerticallyDown() = currentWebView.canScrollVertically(1)
 
     companion object {
+
+        internal const val MAX_JS_ALERT: Int = 2
```

Comment:
I feel like this needs some kind of timer to measure the time between alerts. `3` alerts are fine too if they for example are triggered by user actions. However X alerts in Y seconds is almost always something you do not want. Maybe it's worth exploring the Gecko internals to see how the flag gets set there?

Reply:
Yes, Gecko is using a time-based implementation, I was trying to find the code that does that but was a daunting path, I will give it a second chance.

Reply:
Many `.jsm`,  `.h` and `.cpp` files :P

Reply:
Heh, yeah, that's how it feels to work on Fennec at times. If you do not know about it yet, I mostly read the code via this page: https://searchfox.org/

Reply:
Thanks, Now I know that I'm not alone :)

Reply:
[I found it ](https://dxr.mozilla.org/mozilla-central/source/dom/base/nsGlobalWindowInner.cpp#2026) 🎉 
Working on a similar solution on kt :)